website/content/docs/operations/stateful-workloads.mdx

```diff
@@ -0,0 +1,246 @@
+| [Filestore](https://cloud.google.com/filestore)                    | Zonal/Regional file storage that can be available to multiple instances                                           | Strongly persistent, with an independent lifecycle                               | [Configurable](https://cloud.google.com/filestore/docs/performance)     | File storage that needs to be available to multiple VMs in multiple zones (even only as a failover)                          |
+
+
+### Private cloud or on-premises
+
+When running workloads on-premises in a self-managed private cloud, SAN/NAS
+systems or Software Defined Storage like Portworx or Ceph usually provide
+non-local storage. Compute instances can access the storage using a block
+protocol such as iSCSI, FC, NVMe-oF, or a file protocol such as NFS, CIFS, or
+both. Dedicated storage teams manage these systems in most organizations.
+
+## Consuming persistent storage from Nomad
+
+Since environments differ depending on application requirements, consider
+performance, reliability, availability, and maintenance when choosing the most
+appropriate storage driver.
+
+### CSI
+
+[Container Storage
+Interface](https://github.com/container-storage-interface/spec) is a
+vendor-neutral specification that allows storage providers to develop plugins
+that orchestrators such as Nomad can use. Some CSI plugins can dynamically
+provision and manage volume lifecycles, including snapshots, deletion, and
+dynamic resizing. The exact feature set each plugin supports will depend on the
+plugin and the underlying storage platform.  
+
+Find a list of plugins and their feature set in the [Kubernetes CSI Developer
+Documentation](https://kubernetes-csi.github.io/docs/drivers.html).
+
+While Nomad follows the CSI specification, some plugins may implement
+orchestrator-specific logic that makes them incompatible with Nomad. You should
+validate that your chosen plugin works with Nomad before using it. Refer to the
+plugin documentation from the storage provider for more information.
+
+There are three CSI plugin subtypes:
+
+- **Controller**: Communicates with the storage provider to manage the volume
+	lifecycle. - **Node**: Runs on all Nomad clients and handles all local
+	operations (for example, mounting/unmounting volumes in allocations). The
+	node must be `privileged` to perform those operations. - **Monolithic**:
+	Combines both the above roles.
+
+All types can and should be run as Nomad jobs - `system` jobs for Node and
+Monolithic, `service` for Controllers. More information can be found on the CSI
+concepts [documentation page](/nomad/docs/concepts/plugins/csi).
+
+CSI plugins are useful when storage requirements are quickly and constantly
+evolving due to their dynamic nature - for instance, new workloads that
+frequently need persistent storage added or removed. However, they present some
+challenges in terms of maintenance - most notably, they need to run constantly,
```

Comment:
Let's convert these `-` to em-dashes or get rid of them entirely and break up these run-on sentences.